src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -62,6 +62,23 @@ export const handleEmailSubmission: RequestHandler<
     return res.status(statusCode).json({ message: errorMessage })
   }
 
+  // Check that the form has not reached submission limits
+  const formSubmissionLimitResult = await FormService.checkFormSubmissionLimitAndDeactivateForm(
+    form,
+  )
+  if (formSubmissionLimitResult.isErr()) {
+    logger.warn({
+      message:
+        'Attempt to submit form which has just reached submission limits',
+      meta: logMeta,
+      error: formSubmissionLimitResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(
+      formSubmissionLimitResult.error,
+    )
+    return res.status(statusCode).json({ message: errorMessage })
+  }
```

Comment:
this should be performed after the captcha validation takes place to protect from programmatic submissions, for both email and storage modes.